app/build.gradle

```diff
@@ -264,9 +264,11 @@ android.applicationVariants.all { variant ->
 
         variant.outputs.each { output ->
             def abi = output.getFilter(OutputFile.ABI)
+            // If it is a Mozilla Online build, use a unified version code of armeabi-v7a
+            def arch = (project.hasProperty("mozillaOnline") || gradle.hasProperty("localProperties.mozillaOnline")) ? "armeabi-v7a" : abi
             // We use the same version code generator, that we inherited from Fennec, across all channels - even on
             // channels that never shipped a Fennec build.
-            def versionCodeOverride = Config.generateFennecVersionCode(abi)
+            def versionCodeOverride = Config.generateFennecVersionCode(arch)
 
             println("versionCode for $abi = $versionCodeOverride")
```

Comment:
Suggestion to also print `$abi` and `$mozillaOnline` here - that will make make it easier to debug issues and will also tell the whole truth of what happened. Otherwise it will print an in correct message.

Reply:
I updated the PR to add the `$isMozillaOnline` value to the print statement, also I put a PR on `main` to include the same functionality  https://github.com/mozilla-mobile/fenix/pull/21092

Reply:
The amid of the PR is to have the same `versionCodeOverride` on each apk independently of device  architectures (armeabi-v7a ... x86_64),  based on @rxumoz ticket https://github.com/mozilla-mobile/fenix/issues/20881, application stores in China required that this version must be the same for each apks. The PR tries to  pass a constant (`armeabi-v7a`)  to  [generateFennecVersionCode](https://github.com/mozilla-mobile/fenix/blob/512ee2957cd70e11aacd858ca41e1a1afe2bd09a/buildSrc/src/main/java/Config.kt#L75), as it generates a different `versionCodeOverride` per architecture (armeabi-v7a ... x86_64), passing the same architecture will produce the same version for all apks, this only applies to Mozilla only builds, as for other builds we are going to keep the previous behavior.

Reply:
We are merging to unblock the RC, I think it's a low risk. Please let me know, if there is any comments pending, we can address it on a follow up.

Reply:
It is a bit of a hack that we can probably do in a nicer way, but I agree it is low risk.